### DISCUSSION TITLE
AC modules question

### DISCUSSION THREAD

**Discussion by bkmangx1** [trickerer/Trinity-Bots#328] [Q&A] [answered]
Hey, I'm using some modules on AC that rely on `OnCreatureKilled` and `OnCreatureKilledByPet` hooks. As you can imagine, these do not trigger when bot gets the killing blow. Do you have any suggestions how to check if npcbot was the killer to make these modules work?

**Comment by trickerer** [comment, marked as answer]
You cannot do that inside a module, these additional checks need to be added to Unit.cpp directly. I think I'll just add those myself.

There was a discussion in the past about this, but for TC: #199

**Reply by trickerer** [reply]
I think `OnCreatureKilled` hook is enough. Argument list is the same. Also, both hooks should not process the same kill event to prevent possible double reward or whatever implemented in the module. Another thing is that `OnCreatureKilled` will fall prefectly into current bot kill system where player is always rewarded with loot/xp for creatures killed by bots even if player didn't deal any damage. Plus, pet kill condition exists to trigger spell procs of owner's auras, this shouldn't happen with bots.

**Reply by bkmangx1** [reply]
Yes, the pet one is not very important, just having `OnCreatureKilled` is enough to make these modules function. Thank you for implementing this.
